src/init.cc

```diff
@@ -127,7 +127,7 @@ static ncclResult_t ncclInit() {
         FILE* file;
         if ((file = fopen("/proc/cmdline", "r")) != NULL) {
           if (feof(file) == 0 && ferror(file) == 0) {
-            int len = fread(strValue, 1, 1024, file);
+            int len = fread(strValue, 1, 2047, file);
```

Comment:
is reading 2047 (instead of 2048) bytes on purpose?

Reply:
yes, last byte is reserved for NULL character if valid data reaches 2047.